### DISCUSSION TITLE
Is there a way to set the root directory to the directory of the currently focused buffer?

### DISCUSSION THREAD

**Discussion by dpetka2001** [nvim-tree/nvim-tree.lua#1932] [Q&A] [answered]
Sometimes i open a file with nvim like `nvim ~/.config/nvim/lua/test.lua`.
When I open nvim-tree it opens with the root_dir set at `~` but it correctly focuses and selects the opened file
as I've set in `update_focused_file`. In the manual it says about `update_focused_file` that `update_root` prefers
vim's `cwd` and `root_dirs` otherwise it falls back to the folder containing the file. Is there a way to make the default
behavior when opening nvim-tree, so that the root directory will be that of the folder containing the file? So instead of 
it being set to `~` (from where i run nvim to open the file), it should be set to `~/.config/nvim/lua/` which is the actual 
folder the file resides in? Thanks in advance.

**Comment by alex-courtis** [comment, marked as answer]
You can try something like this:

```lua
local function my_startup(data)

  -- is the buffer a real file?
  if vim.fn.filereadable(data.file) ~= 1 then
    return
  end

  -- is it in a real directory
  local dir = vim.fn.fnamemodify(data.file, ":h")
  if vim.fn.isdirectory(dir) == 0 then
    return
  end

  -- change directory
  vim.cmd.cd(dir)
end

vim.api.nvim_create_autocmd({ "VimEnter" }, { callback = my_startup })
```

See https://github.com/nvim-tree/nvim-tree.lua/issues/1669#issuecomment-1374708397 for more ideas.
